src/components/TeamMemberTasks/TeamMemberTasks.jsx

```diff
@@ -81,47 +89,72 @@ const TeamMemberTasks = props => {
   }, [currentUserId]);
 
   useEffect(() => {
-    renderTeamsList();
-  }, []);
-
-  useEffect(() => {
-    submitTasks();
-    if (userId !== props.auth.user.userid) {
-      dispatch(fetchTeamMembersTask(userId, props.auth.user.userid));
-      const currentUserRole = getUserRole(userId)
-        .then(resp => resp)
-        .then(user => {
-          setUserRole(user.data.role);
-        });
-    } else {
-      dispatch(fetchTeamMembersTask(userId, null));
-      setUserRole(props.auth.user.role);
+    if (isLoading === false && shouldRun) {
+      renderTeamsList();
+      setClickedToShowModal(false);
+      closeMarkAsDone();
+      setCurrentUserId('');
     }
-  }, [updatedTasks]);
+  }, [usersWithTasks, shouldRun]);
+
+  // useEffect(() => {
+  //   //As this useEffect should not run if it's the first run of the page, it will see if the isFirstRun ref is true, in case it is it will change it to false and return
+  //   if (isFirstRun.current) {
+  //     isFirstRun.current = false;
+  //     return;
+  //   }
+  //   submitTasks();
+  //   if (userId !== props.auth.user.userid) {
+  //     dispatch(fetchTeamMembersTask(userId, props.auth.user.userid, false));
+  //     const currentUserRole = getUserRole(userId)
+  //       .then(resp => resp)
+  //       .then(user => {
+  //         setUserRole(user.data.role);
+  //       });
+  //   } else {
+  //     dispatch(fetchTeamMembersTask(userId, null, false));
+  //     setUserRole(props.auth.user.role);
+  //   }
+  // }, [updatedTasks]);
```

Comment:
Is there a reason for commented out code here? If not please remove it.

Reply:
Done